### PR TITLE
Emulator: add web interface virtual mode

### PR DESCRIPTION
**Pull request by RuffaloLavoisier**
Add Web-interface window.

### CODE REVIEW

**Review by RuffaloLavoisier — COMMENTED** [review]
As a result, basically, LGTM

File: emulator/src/Emulator.cpp [hunk 1]
```diff
@@ -9,11 +9,14 @@
...
 #include "../include/Emulator.hpp"
 
+#include "osw_config.h"
+#include "osw_config_keys.h"
+
 OswEmulator* OswEmulator::instance = nullptr;
 
 OswEmulator::OswEmulator() {
     // Init the SDL window and renderer
-    this->mainWindow = SDL_CreateWindow("OSW-OS Emulator", SDL_WINDOWPOS_CENTERED, SDL_WINDOWPOS_CENTERED, DISP_W + this->guiPadding + this->guiWidth + this->guiPadding, DISP_H, SDL_WINDOW_RESIZABLE | SDL_WINDOW_ALLOW_HIGHDPI);
```
Comment (simonmicro) [inline emulator/src/Emulator.cpp:16]: Please revert this. This was intended

File: emulator/src/Emulator.cpp [hunk 2]
```diff
@@ -37,7 +40,13 @@ OswEmulator::~OswEmulator() {
 
     SDL_DestroyWindow(this->mainWindow);
 }
-
+void OswEmulator::drawWatchOverlay() {
```
Comment (simonmicro) [inline emulator/src/Emulator.cpp:43]: What is this doing?

File: emulator/src/Emulator.cpp [hunk 3]
```diff
@@ -82,7 +91,7 @@ void OswEmulator::run() {
 
         // Update the window now with the content of the display
         SDL_RenderPresent(this->mainRenderer);
-
+        // drawWatchOverlay();
```
Comment (simonmicro) [inline emulator/src/Emulator.cpp:94]: And its commented!?

File: emulator/src/Emulator.cpp [hunk 4]
```diff
@@ -135,6 +144,19 @@ void OswEmulator::addGUIHelp(const char* msg) {
         ImGui::EndTooltip();
     }
 }
+char* OswEmulator::convertDateformat(int emulatorWebInterfaceDateFormat) {
```
Comment (simonmicro) [inline emulator/src/Emulator.cpp:147]: Didn't we just added this? We may be able to strip that after the rebase

File: emulator/include/Emulator.hpp [hunk 5]
```diff
@@ -46,6 +54,75 @@ class OswEmulator {
...
+    int inputintSettingDisplayBrightness = DISPLAY_BRIGHTNESS;
+    int inputintSettingDisplayTimeout = DISPLAY_TIMEOUT;
+    int inputintRaiseToWakeSensitivity = WAKE_FROM_RAISE_SENSITIVITY;
+    int inputintTimeZone = CONFIG_TIMEZONE;
+    int inputintStepsPerDay = STEPS_PER_DAY;
+
+    // input editor (float) section
+    float inputfloatDaylightOffset = CONFIG_DAYLIGHTOFFSET;
+
+    void* keyList[25] = {
```
Comment (simonmicro) [inline emulator/include/Emulator.hpp:98]: This makes extension pretty hard... Why?

**Review by simonmicro — CHANGES_REQUESTED** [review]

File: emulator/include/Emulator.hpp (hunk 5 above)
Comment (simonmicro) [reply emulator/include/Emulator.hpp:98]: It draws an overlay, which shows a display border... We should do that in SDL directly - as this is a job of the emulator.

File: emulator/src/Emulator.cpp (hunk 1 above)
Comment (simonmicro) [reply emulator/src/Emulator.cpp:16]: It draws an overlay, which shows a display border... We should do that in SDL directly - as this is a job of the emulator.

**Review by simonmicro — APPROVED** [review]

**Pull request merged**